eth/polls/pollblocks.go

```diff
@@ -35,25 +38,36 @@ func (p *EthBlockPoll) Poll(
 	if p.lastBlock+1 > uint64(state.Block().Height) {
 		return p, nil, nil
 	}
-	lastBlock, results, err := getBlockHashes(p.blockStore, state, p.lastBlock)
+	toBlock := uint64(state.Block().Height)
+	if toBlock-p.lastBlock > p.maxBlockRange {
+		toBlock = p.lastBlock + p.maxBlockRange
+	}
+	lastBlock, results, err := getBlockHashes(p.blockStore, toBlock, p.lastBlock)
 	if err != nil {
 		return p, nil, nil
 	}
 	p.lastBlock = lastBlock
 	return p, eth.EncBytesArray(results), err
 }
 
+// AllLogs pull logs from last N blocks limited by p.maxBlockRange
 func (p *EthBlockPoll) AllLogs(
 	state loomchain.ReadOnlyState, id string, readReceipts loomchain.ReadReceiptHandler,
 ) (interface{}, error) {
-	_, results, err := getBlockHashes(p.blockStore, state, p.startBlock)
+	// Only pull logs from latest height - maxBlockRange
+	toBlock := uint64(state.Block().Height)
+	startBlock := p.startBlock
+	if toBlock-startBlock > p.maxBlockRange {
```

Comment:
I think this should be
```golang
if toBlock-(startBlock+1) > p.maxBlockRange {
```
because blocks will be read starting from `startBlock+1`